src/lib/network_peer/envelope.ml

```diff
@@ -0,0 +1,66 @@
+open Core
+
+module Sender = struct
+  type t = Local | Remote of (Unix.Inet_addr.Blocking_sexp.t * Peer.Id.t)
+  [@@deriving sexp, compare]
+
+  let of_peer (p : Peer.t) = Remote (p.host, p.peer_id)
+
+  let equal sender1 sender2 = Int.equal (compare sender1 sender2) 0
```

Comment:
You can derive equal

Reply:
I cannot, `Unix.Inet_addr` modules do not define `equal`.